walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -159,39 +161,30 @@
   private void findAndAddGapsInOccupiedTimeRanges(long requestedDuration,
       List<TimePoint> orderedTimePoints,
       Collection<TimeRange> availableTimeRanges) {
-    int index = 0;
-    int startCounter = 0;
-    int endCounter = 0;
+    int index = 1;
+    int unendedTimeRange = 1;
 
     while (index < orderedTimePoints.size()) {
-      // Start a continuously occupied range of time.
-      if (startCounter == 0 && orderedTimePoints.get(index).isTimeRangeStart()) {
-        startCounter++;
-        index++;
-      }
-      // Find a continuously occupied range of time, which needs to be skipped. Such continuously
-      // occupied range of time would contain an equal number of start and end {@code TimePoint}.
-      // This occupied range may look like (TR stands for {@code TimeRange}):
+      // Construct a continuously occupied range of time, which needs to be skipped. Such
+      // continuously occupied range of time would contain an equal number of start and end
+      // {@code TimePoint}. This occupied range may look like (TR stands for {@code TimeRange}):
       //    [start of TR1, start of TR2, end of TR1, start of TR3, end of TR3, end of TR2]
-      while (endCounter < startCounter) {
+      // Available time occurs after the continuously occupied ranges of time.
+      if (unendedTimeRange > 0) {
         if (orderedTimePoints.get(index).isTimeRangeStart()) {
-          startCounter++;
+          unendedTimeRange++;
         } else {
-          endCounter++;
+          unendedTimeRange--;
         }
```

Comment:
This could probably be done in one line with a ternary:
`orderedTimePoints.get(index).isTimeRangeStart() ? unendedTimeRange++ : unendedTimeRange--;`

Reply:
It seems that the ternary can't be viewed as a statement. I ended up doing this: 
```
int _ = orderedTimePoints.get(index).isTimeRangeStart() ?
    unendedTimeRange++ : unendedTimeRange--;
```